src/plugins/shared_ux/public/components/exit_full_screen_button/exit_full_screen_button.component.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { MouseEventHandler, HTMLAttributes } from 'react';
+import {
+  EuiScreenReaderOnly,
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiText,
+  useEuiTheme,
+  makeHighContrastColor,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { css } from '@emotion/react';
+
+const label = i18n.translate('sharedUX.exitFullScreenButton.exitFullScreenModeButtonAriaLabel', {
+  defaultMessage: 'Exit full screen mode',
+});
+
+const text = i18n.translate('sharedUX.exitFullScreenButton.exitFullScreenModeButtonText', {
+  defaultMessage: 'Exit full screen',
+});
+
+const description = i18n.translate('sharedUX.exitFullScreenButton.fullScreenModeDescription', {
+  defaultMessage: 'In full screen mode, press ESC to exit.',
+});
+
+/**
+ * Props for the Exit Full Screen button component.
+ */
+export interface Props extends Pick<HTMLAttributes<HTMLDivElement>, 'className'> {
+  onClick: MouseEventHandler<HTMLButtonElement>;
+}
+
+/**
+ * A presentational component that renders a button designed to exit "full screen" mode.
+ */
+export const ExitFullScreenButton = ({ onClick, className }: Props) => {
+  const { euiTheme } = useEuiTheme();
+  const { colors, size } = euiTheme;
+
+  const textCSS = css`
+    ${makeHighContrastColor(colors.mediumShade)(colors.fullShade)};
+  `;
+
+  const buttonCSS = css`
+    &.euiButton {
```

Comment:
@clintandrewhall Is there any way **not** to target `.eui` classes? This really shouldn't be necessary and should be applying custom class names instead like the original Sass did: https://github.com/elastic/kibana/blob/main/src/plugins/kibana_react/public/exit_full_screen_button/_exit_full_screen_button.scss

In EUI, we don't consider changes to internal classes to be breaking changes and when we do (and could very soon) this will break your styles.

Reply:
As we discussed, I'm opening a follow-up PR to address this, and will follow up with EUI to document issues.  Thanks for your help here, @cchaos !!